intro/doodle/algobase.h

```diff
@@ -0,0 +1,476 @@
+// -*- C++ -*-
+
+// Copyright (C) 2007-2023 Free Software Foundation, Inc.
```

Comment:
We should not need to import this file. If it's included directly, try including `<algorithm>` instead.